Add Continuous Optimization Controls to BotorchRecommender (#389)
emdgroup/baybe: docs/userguide/recommenders.md

* The **[`BotorchRecommender`](baybe.recommenders.pure.bayesian.botorch.BotorchRecommender)**
  is a powerful recommender based on BoTorch's optimization engine that can be applied
  to all kinds of search spaces. In continuous spaces, its `sequential_continuous` flag
  allows to choose between greedy sequential optimization and batch optimization as the
  underlying point generation mode. In discrete/hybrid spaces, sequential greedy
  selection is the only available mode and is thus activated automatically.
  
  Note that the recommender performs a brute-force search when applied to hybrid search
  spaces, as it does gradient-based optimization in the continuous part of the space

>   while exhaustively searching choices in the discrete subspace. You can customize this

Reviewer: ```suggestion
  while exhaustively evaluating configurations of the discrete subspace. You can customize this
```

  behavior to only sample a certain percentage of the discrete subspace via the
  `sample_percentage` attribute and to choose different sampling algorithms via the

>   `hybrid_sampler` attribute. The gradient-based optimization part can also further be

Author: changed

>   controlled by the `num_restarts` and `raw_samples` keywords, for details please refer

Reviewer: connecting two main clauses with comma

Author: changed
